practices/libraries.md

```diff
@@ -0,0 +1,99 @@
+- While using the library, the program gains the library’s dependencies and does not have to implement the dependencies itself.
+- The library’s properties can be connected to the program in any lifecycle phase.
+
+## Why You Might Want the Libraries to Be Updated
+
+Sometimes the updates are arriving almost every day and it can get very annoying. Updating is a pain. It takes a long time and after the update, nothing works as before. But remember - libraries updates might be very important. The dependencies from the specific library you need for your program to be working may change.
+
+You should keep track of those updates, they might contain important changes, for example:
+
+- Security vulnerabilities fixes
+- Bug fixes
+- Performance improvements
+- New functionalities
+- Better user interface
+
+You might need some of these changes urgently. Checking them regularly and updating them as soon as possible saves you time and money. You have to update the dependencies anyway, so why to postpone and, therefore, accumulate a maintenance debt?
+
+## Problems the Updated Libraries Solve
+
+- [Increased cost](/problems/increased-cost)
+- [Poor Code Quality](/problems/poor-code-quality)
+- [Meaningless Work](/problems/meaningless-work)
+- Developers vs. Operations wars
+- [Long Feedback Loops](/problems/long-feedback-loops)
+- Unhappy Clients
+
+## How to Implement the Updating of Libraries
+
+There are two strategies to implement the regular updates:
+
+1. A common strategy is to install **only the most necessary updates**, such as security fixes. This strategy saves time you would spend with installing all the updates. A lot of developers tend to update as little changes as possible, or they bundle them together to do it just a few times per year. But this strategy brings risks of missing some important updates.
+
+2. Extreme Programming agile approach suggests to **update as much as possible, as often as possible**. This strategy requires a high level of automation - otherwise, the developers become just very expensive updating machines and they soon lose interest in their jobs.
+
+What can help you with the updates automating? Automate these:
+
+- builds
+- integration
+- assurance
+- testing
+- releases
+- [deployment](/practices/automated-deployment)
+
+Use continuous delivery and continuous deployment. The dependencies updates can be integrated directly into your program delivery pipeline. There are tools that check automatically if there are some new dependencies updates, such as [npm check updates](https://github.com/tjunnone/npm-check-updates) for JavaScript. Npm check updates even upgrades your package.json dependencies to the latest versions.
+
+The second strategy is more expensive than the first one but it lowers the risks. Automation of the updates is probably the best solution.
+
+## Common Pitfalls of Updating the Libraries
+
+1. Developers ignore updating the dependencies because they think it would be just a waste of time.
```

Comment:
```suggestion
- Developers ignore updating the dependencies because they think it would be just a waste of time.
- Not automatized library updates.
```